Source/Fuse.Reactive.JavaScript/ModuleInstance.uno

```diff
@@ -0,0 +1,110 @@
+using Uno;
+using Uno.Collections;
+using Fuse.Scripting;
+
+namespace Fuse.Reactive
+{
+	class ModuleInstance: DiagnosticSubject
+	{
+		readonly ThreadWorker _worker;
+		readonly JavaScript _js;
+		readonly Dictionary<string, object> _deps = new Dictionary<string, object>();
+
+		// UI thread
+		public ModuleInstance(ThreadWorker worker, JavaScript js)
+		{
+			for (var i = 0; i < js.Dependencies.Count; i++)
+				_deps.Add(js.Dependencies[i].Name, worker.Unwrap(js.Dependencies[i].Value));
+
+			_js = js;
+			_worker = worker;
+			_worker.Invoke(Evaluate);
+		}
+
+		// JS thread
+		void Evaluate()
+		{
+			_js.ScriptModule.Dependencies = _deps;
+			_dc = _worker.Reflect(EvaluateExports());
+			UpdateManager.PostAction(SetDataContext);
+		}
+
+		object _dc;
+
+		// UI thread
+		void SetDataContext()
+		{
+			_js.SetDataContext(_dc);
+		}
+
+		ModuleResult _moduleResult;
+
+		public void Dispose()
```

Comment:
Dipose needs to somehow block `Evaluate` from posting results. With the dependencies it's possible to have the same module evaluating in parallel multiple times if the deps change quickly. Only the final, current, evaluation of the module will be expected.